python/ray/tune/trial_runner.py

```diff
@@ -108,6 +108,10 @@ class TrialRunner:
             If fail_fast='raise' provided, Tune will automatically
             raise the exception received by the Trainable. fail_fast='raise'
             can easily leak resources and should be used with caution.
+        rerun_failed (bool): Resets and reruns failed trials, assuming
+            the provided Trainable is the same. Previous trial artifacts
+            will be left untouched. Only to be used with
+            `resume` enabled. Raises ValueError otherwise.
```

Comment:
One thing that was confusing for me is that it only reruns previously failed trials. At first I thought it would rerun failing trials as soon as they fail (potentially ending in an infinite loop)